R/pkg/tests/fulltests/test_sparkSQL_arrow.R

```diff
@@ -68,7 +68,7 @@ test_that("createDataFrame/collect Arrow optimization - type specification", {
     callJMethod(conf, "set", "spark.sql.execution.arrow.sparkr.enabled", arrowEnabled)
   })
 
-  expect_equal(collect(createDataFrame(rdf)), expected)
+  expect_true(all(collect(createDataFrame(rdf)) == expected))
```

Comment:
Here I work around to make the tests work with any R version. The problem was [R 4.1 introduced `check.tzone` at `all.equal`](https://cran.r-project.org/doc/manuals/r-release/NEWS.html) which apparently `testthat` uses.

When you collect from a POSIXct with an empty tzone (by default), Arrow conversion fills a local timezone instead of being empty:

```r
rdf <- data.frame(list(list(t = as.POSIXct("1990-02-24 12:34:56", tz="UTC"))))
SparkR:::callJMethod(SparkR:::callJMethod(spark, "conf"), "set", "spark.sql.execution.arrow.sparkr.enabled", "false")
withoutArrow <- collect(createDataFrame(rdf))
SparkR:::callJMethod(SparkR:::callJMethod(spark, "conf"), "set", "spark.sql.execution.arrow.sparkr.enabled", "true")
withArrow <- collect(createDataFrame(rdf))

attr(withoutArrow$t, "tzone")
attr(withArrow$t, "tzone")
```
```
[1] ""
[1] "Asia/Seoul"
```

 Spark returns a local time instances in Scala, Python and R. Therefore, I think either being an empty timezone or local timezone can be correct in Spark context, and it's not an issue IMO.

Reply:
cc @MaxGekk FYI